CLUE_Dice_Roller/clue_dice_roller.py

```diff
@@ -0,0 +1,108 @@
+"""
+Dice roller for CLUE
+
+Set the number of dice with button A (1-2-3-4-5-6)
+and the type with button B (d4-d6-d8-d10-d12-d20-d100).
+Roll by shaking.
+Pressing either button returns to the dice selection mode.
+"""
+
+import time
+import board
+from random import randint
+from adafruit_clue import clue
+from adafruit_debouncer import Debouncer
+import displayio
+from adafruit_bitmap_font import bitmap_font
+import terminalio
+from adafruit_display_text import label
+
+
+# input constraints
+MAX_NUMBER_OF_DICE = 6
+SIDES = [4, 6, 8, 12, 20, 100]
+
+# modes: selecting/result
+SELECTING = 0
+ROLL_RESULT = 1
+
+number_of_dice = 0                        # 0-relative, gets adjusted to 1-relative before display/use
+side_selection = 0
+
+button_a = Debouncer(lambda: clue.button_a)
+button_b = Debouncer(lambda: clue.button_b)
+
+
+# Set up display
+
+display = board.DISPLAY
+selection_font = bitmap_font.load_font('/Helvetica-Bold-36.bdf')
+selection_font.load_glyphs(b'0123456789XDd')
+selection_color = 0x0000FF
+
+roll_font = bitmap_font.load_font('/Anton-Regular-104.bdf')
+roll_font.load_glyphs(b'0123456789XDd')
+roll_color = 0xFFFFFF
+
+selection_label = label.Label(selection_font, x=0, y=25, text='XdXXX', color=selection_color)
+roll_label = label.Label(roll_font, x=0, y=150, text='XXX', color=roll_color)
+
+group = displayio.Group()
+group.append(selection_label)
+group.append(roll_label)
+
+display.show(group)
+
+# Helper functions
+
+def roll(count, sides):
+    for i in range(15):
+        roll = sum([randint(1, sides) for d in range(count + 1)])
+        roll_label.text = str(roll)
+        roll_label.x = 120 - (roll_label.bounding_box[2] // 2)
+        duration = (i * 0.05) / 2
+        clue.play_tone(2000, duration)
+        time.sleep(duration)
+
+
+def update_display(count, sides):
+    selection_label.text = '{0}d{1}'.format(count + 1, SIDES[sides])
+    selection_label.x = 120 - (selection_label.bounding_box[2] // 2)
+    roll_label.text = ''
+
+
+def reset_selection_screen():
+    global number_of_dice, side_selection
+    number_of_dice = 0
+    side_selection = 0
+    update_display(number_of_dice, side_selection)
+
+
+
+display
+mode = SELECTING
+update_display(number_of_dice, side_selection)
+
+while True:
+    button_a.update()
+    button_b.update()
+
+    if mode == SELECTING:
+        if button_a.rose:
```

Comment:
i think people expect it to change on falling (pressed) not rising?

Reply:
The clue button properties return True when pressed.  As does the CPB buttons.  Both those surprised me for just that reason.